agir/msgs/actions.py

```diff
@@ -1,27 +1,125 @@
+from django.db import connection
 from django.db.models import (
     OuterRef,
-    Case,
-    When,
-    Count,
     Subquery,
-    IntegerField,
-    F,
     Exists,
     Q,
     Max,
     DateTimeField,
-    Prefetch,
+    F,
+    Count,
+    Value,
 )
 from django.db.models.functions import Greatest, Coalesce
 
-from agir.groups.models import SupportGroup, Membership
+from agir.activity.models import Activity
+from agir.groups.models import Membership
 from agir.msgs.models import (
     SupportGroupMessageRecipient,
     SupportGroupMessage,
     SupportGroupMessageComment,
 )
+from agir.notifications.models import Subscription
 from agir.people.models import Person
 
+RECENT_COMMENT_LIMIT = 4
+
+# Requête alternative : le nombre de messages qui ont au moins un commentaire nouveau / non utilisé pour le moment
+MESSAGES_WITH_UNREAD_COMMENTS_COUNT_REQUEST = """
+SELECT COUNT(*) FROM msgs_supportgroupmessage message
+ LEFT JOIN msgs_supportgroupmessagerecipient recipient ON recipient.message_id = message.id
+ JOIN groups_membership membership ON message.supportgroup_id = membership.supportgroup_id 
+ WHERE recipient.recipient_id = '74178da8-d8d9-422e-b16f-6aa2b253a9de'
+ AND membership.person_id = '74178da8-d8d9-422e-b16f-6aa2b253a9de'
```

Comment:
Est-ce qu'il faut faire comme ça ? 
```suggestion
 WHERE recipient.recipient_id = %(person_id)s
 AND membership.person_id = %(person_id)s
```

Reply:
Oui ! Après on pourrait virer toute la requête en vrai, j'avais commencé à bosser là-dessus au départ en pensant qu'on voulait calculer le nombre de messages avec une partie non lue, plutôt que le nombre de commentaires non lus.